cibuildwheel/macos.py

```diff
@@ -409,7 +409,9 @@ def build(options: Options, tmp_path: Path) -> None:
                                 config.version
                             )
                         )
-                        build_env["PIP_CONSTRAINT"] = constraint_path.as_uri()
+                        build_env["PIP_CONSTRAINT"] = (
+                            constraint_path.as_uri() + " " + build_env.get("PIP_CONSTRAINT", "")
+                        )
```

Comment:
Hmm, I'm not sure. In this case wouldn't it be better to tell the user to set `CIBW_DEPENDENCY_VERSIONS=latest` instead? Or set it to your own lockfile? It feels a bit like a foot-gun if setting PIP_CONSTRAINT removes the build determinism feature from cibuildwheel.

Reply:
Well, that option isn't exactly what you're after I think. That option is used by cibuildwheel to pin versions of build tools on Mac and Windows. It isn't used on Linux builds, because there, the versions of tools is instead governed by which manylinux image is used.

I guess this is why I'm keen for [a more first-party supported way](https://github.com/pypa/build/issues/292) to do this - as we could specifically target the versions of build-backends, not the surrounding tools like pip or virtualenv.

I'm now thinking... maybe the better answer is for cibuildwheel to stop including setuptools in our constraint file. We don't pin other build backends, and the platform inconsistency above means that the setuptools pin is used inside mac & windows isolated builds, but not on linux (where I suppose the latest is fetched by `build` when it sets up the build venv).

Reply:
> maybe the better answer is for cibuildwheel to stop including setuptools in our constraint file

On reflection, this does seem like a good idea. The only issue with this idea is that projects without pyproject.toml are gonna lose our pin. But projects **with** pyproject.toml aren't getting their build-backend pinned anymore anyway, so it's not a huge issue.

I think we'd also want to remove the auto-install of setuptools and wheel. That feels a bit legacy at this point, given that setuptools is no longer the only game in town, and that most(?) projects are on pyproject.toml.

I'm currently putting together a PR for this, let's see how it looks...

Reply:
Just pushed the above in PR #1725. But, at the same time, if you're doing your own pinning via PIP_CONSTRAINT you can just disable cibuildwheel's pinning using `CIBW_DEPENDENCY_VERSIONS=latest`.